lib/galaxy/config.py

```diff
@@ -383,6 +390,14 @@ def __init__( self, **kwargs ):
         self.galaxy_infrastructure_url = galaxy_infrastructure_url
         self.galaxy_infrastructure_url_set = galaxy_infrastructure_url_set
 
+        # Verify host:port is available before continuing
+        try:
+            q = bind(*re.sub(r'http(s?)\://', '', self.galaxy_infrastructure_url).split(':'))
```

Comment:
I don't think there is any guarantee that ``self.galaxy_infrastructure_url`` is even an accessible hostname on the Galaxy server. I think you want to change code in the paste server not in Galaxy. Galaxy doesn't really know what Galaxy is trying to bind to.